### DISCUSSION TITLE
FEC Codes not updated to new one's ? (Solved)

### DISCUSSION THREAD

**Discussion by tommybol** [Mr-MIBonk/M.I.B._More-Incredible-Bash#414] [General]
Hi All,

I have a 2021 Audi TT with MHI_ER_AU43x_P5404_MU1479 installed on my system trying to add CarPlay.
Did everything as described in the tutorial. 
M.I.B. Installed successfully in the red menu, got into green menu and started M.I.B.

1. I made a manual backup.
2. And as far as i could understand the IFS patch was installed successfully. (see screenshot)
![429331515_1485321485382435_608182754937527968_n](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/128286113/4f4734d0-50f6-4a0f-a75d-35524d031b2f)
Only CarPlay did not came on my display. So i manually rebooted the system and it still wasn't there.
3. After this I did the SVM Fix (i know i forgot it the first time) but i don't know if that was an important step, but still no change.
4. I manually added CarPlay and Android Auto in M.I.B. Now the Smartphone Interface Icon showed upon my screen. But still not working.
5. tried to run the IFS patch again, but it said the folder wasn't there anymore, upon inspection it was gone. So i assume the system did this to prevent bricking it. (had to try)

But when checking my FEC's in the Red menu, I noticed that it did not update to the new one's, so i believe my problem is here. (see picture).
![429362459_365259829687199_6707443432832250419_n](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/128286113/182b4cab-84b7-4576-895e-fb17c0dccc98)
No CarPlay or Android codes there.

So i'm not sure if the IFS patch did run and worked ? How can i tell ?
Or what can i do more ? Anyone got a golden answer ? I believe i'm on the right track ?
Don't want to brick my system.

Thank you very much !
Attached is my log file.

Greetings,
Tom
[MU1479-HHA-07327.02.21437002CE.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/14425881/MU1479-HHA-07327.02.21437002CE.log)

**Comment by tommybol** [comment]
A new day, new possibilities :-)
one step closer.

I could find out by comparing the log file with others that the IFS patch did run. 
IFS-stage2.......4512+0 records in 4512+0 records out -> done :-)

So i copied again the M.I.B. Patch to the patches folder.
And ran the add FEC in M.I.B. rebooted the system, and now the system has taken the FEC's and CarPlay worked just once.
-----
timestamp: 1970_01_01_00_01_35

MMX-fecel---->

add FECs to FEC file from M.I.B backup folder
Existing FecContainer parsed
FEC File build done!

copy FEC file from /patches/MHI2_ER_AU43x_P5404_MU1479_PATCH... -> done :-)

-----------
timestamp: 1970_01_01_00_01_41

MMX-fecel---->
copy EL file from /patches/MHI2_ER_AU43x_P5404_MU1479_PATCH... -> done :-)

-----------
timestamp: 1970_01_01_00_01_45

MMX-addfec---->
FECs from /net/rcc/mnt/efs-persist/FEC
FECs currently used by unit
00060800
00060900
023000EE
00040100
02300032
00050000
00070200
00030000

FECs from /net/rcc/mnt/efs-system/backup/FEC
FECs from factory
00040100
02300032
00050000
00070200
00030000
-----
upon checking the FEC Codes i could see they were added but marked as illegal.

Trying to figure out why they are illegal.
![429643210_277937978666095_8420250473158082520_n](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/128286113/b9548da9-85f7-48b0-bba0-b4a15f0caf0c)

Any suggestions are welcome.
Thanks again !

**Comment by tommybol** [comment]
Problem solved, it seemed the system did not run the ifs-patch at all.
MacOs was the problem.
Took a clean SD-card, windows computer formatted it to fat32 and prepared it as described.
Ran SWDL_Cleaner to clean out the illegal FEC's.
Then ran the AIO patch again.

Hope someone might find this usefull.

**Reply by Tomt97** [reply]
do we need to run SWDL_Cleaner? 

do we need to move the patch over our backup. or just run the AIO patch

**Comment by fuzz1234** [comment]
Hi
How much time did it take for you to install that on your car? Trying to install mib toolbox and it's saying 60 minutes required. How did you maintain power to your car? Will the car not go in sleeping mode in less than one hour?

Thanks

**Reply by tommybol** [reply]
Hi, As soon as the SD card was prepped correctly. It took about 10 minutes to install the patch in the car. Overall time is about 60 minutes when you need to prep your sd card etc. I used a battery charger attached to the car when flashing. The system does not go in sleeping mode once it is running the patch installer. 

Good Luck !

**Comment by fuzz1234** [comment]
Hi
I have a small 12 Volts battery charger. Trickle charger. I don't know if that would be enough. What kind of battery charger did you use? What was it's amperage?

**Reply by tommybol** [reply]
i used a ctek mxs 5.0 its  5 Amp. Was more than enough for me.

**Comment by Alex320dd** [comment]
Hab das gleiche Problem in einer a4 b9 bj 2016
![20241114_203132](https://github.com/user-attachments/assets/4405d84a-e7d8-4ee3-837f-1960c179c90b)
![20241114_202913](https://github.com/user-attachments/assets/353a469a-8d6d-4d7a-ad69-ded54e01fed5)
![20241114_202908](https://github.com/user-attachments/assets/ced76d1e-2669-4dfa-b39e-8c973ef2ce52)
![20241114_182511](https://github.com/user-attachments/assets/75e36fa8-677f-4997-968b-4f3a82b5143e)
![20241114_182734](https://github.com/user-attachments/assets/86cc545a-b061-4e75-bba7-0b986276a15b)
![20241114_183009](https://github.com/user-attachments/assets/ae9a51a6-7425-4018-b629-f617836255b1)
![20241114_200450](https://github.com/user-attachments/assets/bd9e314b-e6ff-4f2c-ada8-7dab3765da12)
![20241114_182343](https://github.com/user-attachments/assets/e3caa8e8-90c3-4755-ba12-306ca8e1e279)

**Reply by Alex320dd** [reply]
Brauche hilfe
